crates/bevy_color_blindness/src/lib.rs

```diff
@@ -0,0 +1,412 @@
+///             mode: Mode::Deuteranomaly,
+///             enable: true,
+///         })
+///         // add the plugin
+///         .add_plugin(ColorBlindnessPlugin)
+///         .add_startup_system(setup)
+///         .run();
+/// }
+///
+/// fn setup(mut commands: Commands) {
+///     // set up your scene...
+///
+///     // create the camera
+///     commands
+///         .spawn_bundle(Camera3dBundle {
+///           transform: Transform::from_xyz(-2.0, 2.5, 5.0).looking_at(Vec3::ZERO, Vec3::Y),
+///           ..Default::default()
+///         })
+///         // IMPORTANT: add this component to your main camera
+///         .insert(ColorBlindnessCamera);
+/// }
+/// ```
+///
+/// # Important note
+///
+/// This plugin only simulates how color blind players will see your game.
+/// It does not correct for color blindness to make your game more accessible.
+/// This plugin should only be used during development, and removed on final builds.
+pub struct ColorBlindnessPlugin;
+impl Plugin for ColorBlindnessPlugin {
+    fn build(&self, app: &mut App) {
+        let world = &mut app.world;
+        world.get_resource_or_insert_with(ColorBlindnessParams::default);
+
+        load_internal_asset!(
+            app,
+            COLOR_BLINDNESS_SHADER_HANDLE,
+            "color_blindness.wgsl",
+            Shader::from_wgsl
+        );
+
+        app.add_plugin(Material2dPlugin::<PostProcessingMaterial>::default())
+            .add_startup_system(setup)
+            .add_system(set_camera_target)
+            .add_system(update_percentages);
+    }
+}
+
+/// handle to the color blindness simulation shader
+const COLOR_BLINDNESS_SHADER_HANDLE: HandleUntyped =
+    HandleUntyped::weak_from_u64(Shader::TYPE_UUID, 3937837360667146578);
+
+/// Resource which selects the type of color blindness to simulate
+///
+/// # Example
+///
+/// This system will only enable the simulation while `Space` is held, and will cycle through
+/// the different modes when `N` is pressed.
+///
+/// ```rust
+/// # use bevy_app::prelude::*;
+/// # use bevy_color_blindness::*;
+/// # use bevy_ecs::prelude::*;
+/// # use bevy_input::prelude::*;
+/// # use bevy_math::prelude::*;
+/// # use bevy_transform::prelude::*;
+/// fn cycle_mode(input: Res<Input<KeyCode>>, mut params: ResMut<ColorBlindnessParams>) {
+///     if input.just_pressed(KeyCode::N) {
+///         params.mode.cycle();
+///         println!("Changed to {:?}", params.mode);
+///     }
+///
+///     params.enable = input.pressed(KeyCode::Space);
+/// }
+/// ```
+#[derive(Default, Debug)]
+pub struct ColorBlindnessParams {
+    /// Selects the color blindness mode to use
+    ///
+    /// Defaults to `Mode::Normal`
+    pub mode: Mode,
+    /// Controls whether color blindness simulation is enabled
+    ///
+    /// Defaults to `false`
+    pub enable: bool,
+}
+
+/// The different modes of color blindness simulation supported.
+#[derive(Clone, Default, Debug)]
+pub enum Mode {
```

Comment:
I'd really like one-sentence descriptions of the effect of each of these modes on the enum variants.